x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DataType.java

```diff
@@ -199,7 +200,7 @@ public static DataType fromJava(Object value) {
         if (value instanceof ZonedDateTime) {
             return DATETIME;
         }
-        if (value instanceof String || value instanceof Character) {
+        if (value instanceof String || value instanceof Character || value instanceof BytesRef) {
```

Comment:
This is the only extra method we needed. There are some things we could remove but I'm not sure that's worth doing. Maybe though! Opinions?

Reply:
There are some things that don't apply (yet?) to ESQL. BigInteger, Short, Byte....

But, the thing I am confused about... what's the general approach when it comes to `esql-core` module: we keep it and make it ESQL specific by removing everything that used to be in the `ql` module AND used by other projects other than ESQL, or we merge it in `esql`. Take for example [@not-napoleon's recent PR](https://github.com/elastic/elasticsearch/pull/109216/files) that moved tests and optimization rules out of `esql-core` and into `esql`.

Reply:
In the long term, I think the strategy is to merge it into `esql`, but in the short term I expect many tactical refactorings in the other direction (like this one), just because it's easier

Reply:
Yeah. I want to move it into `esql` entirely, but I think deduplicating stuff is more important. I think here I'm going to remove `EsqlDataTypes` entirely *eventually*, but that won't get moved into esql for a while. I guess for me job 1 is making stuff simpler and job 2 is moving stuff into esql.